File: parking_lot_system/parking_lot/parking_strategy.py
class ParkingStrategy:
    def find_spot(self, level, vehicle_type):
        raise NotImplementedError

class DefaultParkingStrategy(ParkingStrategy):
    def find_spot(self, level, vehicle_type):
        for spot in level.spots:
            if not spot.is_occupied and spot.spot_type == vehicle_type:
                return spot
        return None

File: parking_lot_system_go/parkinglot/parking_strategy.go
package parkinglot

type ParkingStrategy interface {
	FindSpot(level *ParkingLevel, vehicleType VehicleType) *ParkingSpot
}

type DefaultParkingStrategy struct{}

func (s *DefaultParkingStrategy) FindSpot(level *ParkingLevel, vehicleType VehicleType) *ParkingSpot {
	for _, spot := range level.Spots {
		if !spot.IsOccupied && spot.SpotType == vehicleType {
			return spot
		}
	}
	return nil
}
